### PR TITLE
#790 Part4 : Compile LGAT subtypes

### PR DESCRIPTION
**Pull request by kgaonkar6**
#### What scientific question is your analysis addressing?
Compile LGAT subtypes derived from SNV, CNV,SV and fusion molecular abberations. 

#### What was your approach?
I used files from formatted to each subtype in lgat_subset folder and compiled the subtype for each unique row== Kids_First_Paticipant_ID+ sample_id. 
The  matched DNA ids are in Kids_First_Biospecimen_ID_DNA and RNA ids are in Kids_First_Biospecimen_ID_RNA and the compiled molecular_subtype is summarized in molecular_subtype. 

#### What GitHub issue does your pull request address?
#790 

#### Which areas should receive a particularly close look?
Are all conditions applied for each subtype mentioned in #790? 
I've added NF1 germline predisposition in this notebook since it wasn't a molecular aberration but a clinical file value also I'm keeping "NF1" and "NF2"  cancer_disposition does that seem ok?

#### Is there anything that you want to discuss further?
Maybe #923 needs to be discussed before we can merge this PR

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?
Yes

#### What types of results are included (e.g., table, figure)?
tables

#### What is your summary of the results?
313 unique LGAT sample_ids have subtype compiled in this notebook.

#### Reproducibility Checklist

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has been added to continuous integration.

#### Documentation Checklist

- [x] This analysis module has a `README` and it is up to date.
- [x] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

File: analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd
Comment (jaclyn-taroni) [inline analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd:2]: Update the title to be more descriptive

File: analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd
Comment (jaclyn-taroni) [inline analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd:29]: Should this be using the same inclusion/exclusion criteria as what is used elsewhere and described here: https://github.com/AlexsLemonade/OpenPBTA-analysis/tree/0ed49b68a62462d2c96495ffea1bdadd5b4ff835/analyses/molecular-subtyping-LGAT#inclusionexclusion-criteria ?

File: analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd
Comment (jaclyn-taroni) [inline analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd:93]: Are the `cancer_predispositions` values standardized in some way such that you could do exact matching for both NF1 and NF2?

File: analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd
Comment (jaclyn-taroni) [inline analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd:153]:
```suggestion
OR harbors a MET SNV
```

File: analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd
Comment (jaclyn-taroni) [inline analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd:120]: Will domain experts know that "non-canonical BRAF alterations such as p.V600ins or p.D594N" could fall under the `LGG, BRAF V600E` umbrella?

File: analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd
Comment (jaclyn-taroni) [inline analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd:176]:
```suggestion
OR harbors FGFR1-TACC1 fusion
```

File: analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd
Comment (jaclyn-taroni) [inline analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd:282]: Would you be better off not including `LGG, ` in any of the steps above? Instead you could use `tidyr::unite()` to combine the columns separated by `, ` and then I think you could use `case_when()` with something like (untested):

```R
case_when(
  molecular_subtype == "NA" ~ "LGG, To be classified",
  TRUE ~ stringr::str_c("LGG, ", molecular_subtype)
)
```

File: analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd
Comment (jaclyn-taroni) [inline analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd:275]: Hm, I've never thought about using `summarise()` like this... it's possible [`tidyr::unite()`](https://tidyr.tidyverse.org/reference/unite.html) would give you a bit more control. See other review comment about not including `LGG, `.

**Review by jaclyn-taroni — COMMENTED** [review]
Hi @kgaonkar6 - I like the way you've split out each subtype in this notebook! I had a few questions/comments to get us started.

File: analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd
Comment (kgaonkar6) [reply analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd:2]: haha.. I'll update that!

File: analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd
Comment (kgaonkar6) [reply analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd:29]: Yes I think so, I can add the inclusion/exclusion from the yaml file used before

File: analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd
Comment (kgaonkar6) [reply analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd]: I tried the unite(all_subtype,"molecular_subtype",sep=",") and it collapsed all columns in `all_subtype` it didn't look right, am I missing something? unite() seems to combine col, which seems like it's not applicable here since we have 1 molecular_subtype column.

I used summarise since we need to combine molecular_subtype per unique row of ids (ie. combine if there are 2 rows 1 from all_dna_subtype and another 1 from all_rna_subtype only differing on molecular_subtype)

Comment (kgaonkar6) [reply analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd]: I did implement your suggestion in your comment, that was useful to clean up some code

File: analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd
Comment (kgaonkar6) [reply analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd]: I updated to use specific values related to NF1/NF2 cancer predisposition to do exact matching

File: analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd
Comment (jaclyn-taroni) [reply analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd:295]: Oh I see, I was thinking that there were two columns: one with subtype based on RNA data and one with subtype with DNA data, but these are encoded in different _rows_. That's where the `unite()` suggestion was coming from.

File: analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd
Comment (kgaonkar6) [reply analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd]: hmm using the inclusion/exclusion from the yaml file removes "7316-1106" which was updated to Low grade from path review. 

https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/f85831fe7e405b1c6b6c43274cabcc796b7bd3b1/analyses/molecular-subtyping-pathology/03-incorporate-pathology-feedback.Rmd#L199-L224

Comment (kgaonkar6) [reply analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd]: I think it should be reverted back to broad_histology filter, thoughts?

Comment (jaclyn-taroni) [reply analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd]: My understanding is that `broad_histology` depends on `integrated_diagnosis` (or would depend on in the future), which depends on `molecular_subtype` - I'm trying to avoid circularity.

File: analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd
Comment (kgaonkar6) [reply analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd:140]: In this [LGG review](https://www.sciencedirect.com/science/article/abs/pii/S1535610820301513) they gathered the 2 sites in BRAF p.V600E subtype so we also added to the list for other BRAF SNVs should I link it in the notebook? 

Here's the text from the paper: 
>Together, KIAA1549-BRAF (n = 166), BRAF p.V600E (n = 79), and germline NF1 mutations (n = 79) accounted for 68% (n = 324) of tumors (Figures 2A–2C). Rare alterations accounted for an additional 17% of cases. These included non-canonical BRAF alterations, such as fusions partnered with genes others than KIAA1549 (n = 4), 2 insertion events at position 600 (p.V600ins) and 1 SNV at position 594 (p.D594N) (Figures 2A)
image

Comment (jaclyn-taroni) [reply analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd:140]: I guess I'm wondering if something like "LGG, Other BRAF alteration" would be more appropriate in those cases, but that would probably imply fusions other than the canonical BRAF fusion which is captured by "other MAPK"

Comment (kgaonkar6) [reply analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd:140]: ah right!  Yeah it would be clearer to add the other BRAF snvs to "other MAPK", checking the script for if we have these other snvs in BRAF pV600E subtype

Comment (kgaonkar6) [reply analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd:140]: Seems like in the 01 script I only looked for 599, 600 or SNVs overlapping tyrosine kinase domain: https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/f85831fe7e405b1c6b6c43274cabcc796b7bd3b1/analyses/molecular-subtyping-LGAT/01-subset-files-for-LGAT.Rmd#L146-L156

```r
consensusMutationSubset[grep("BRAF",consensusMutationSubset$Hugo_Symbol),"HGVSp_Short"] %>% unique()
# A tibble: 4 x 1
  HGVSp_Short    
  <chr>          
1 p.V600E        
2 p.T599dup      
3 p.A598_T599insV
4 p.V600= 
```

I'm updating the comment in the notebook for BRAF V600E subtype

File: analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd
Comment (kgaonkar6) [reply analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd:295]: I did try to do left joins with all_dna_subtypes and all_rna_subtypes and then do `unite()` but seems like the NA's get combined as well then would need to check if `NA, NA` OR `NA, <subtype value>` OR `NA, <subtype value>, <subtype value> etc.. in case_when.

I can work on that part further if using `unite` instead of the 2 step `groupby+summarise` would make the code more clear, let me know!

File: analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd
Comment (kgaonkar6) [reply analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd]: I actual forgot to use `pbta-histologies-base.tsv` since it is a moleuclar-subtyping module, I'am updating that.

File: analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd
Comment (jharenza) [reply analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd]: > I guess I'm wondering if something like "LGG, Other BRAF alteration" would be more appropriate in those cases, but that would probably imply fusions other than the canonical BRAF fusion which is captured by "other MAPK"

I had put them in the v600e group per @adamcresnick's request.

Comment (jaclyn-taroni) [reply analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd]: Okay, I think the documentation needs to be very clear about the criteria.

Comment (jharenza) [reply analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd]: Agree!

File: analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd
Comment (jaclyn-taroni) [reply analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd:295]: [Newer versions of `tidyr::unite()` have an `na.rm` argument](https://tidyr.tidyverse.org/reference/unite.html), but I _don't think_ the version we have on the project image is new enough (you might double check that if you have not already!).

I don't know that this is worth spending a lot of time on without the `na.rm` functionality.

File: analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd
Comment (jaclyn-taroni) [reply analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd]: I skimmed the HGG module and the earlier notebook in this module - it seems like `pbta-histologies-base.tsv` and using the JSON file is the way to go. Is that correct?

File: analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd
Comment (jaclyn-taroni) [inline analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd:70]: I take it that no WXS samples are included after the inclusion/exclusion criteria?

File: analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd
Comment (jaclyn-taroni) [inline analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd:112]: Reading up on #790 - I don't see why we should be including NF2 here. Did I miss something in an earlier PR?

**Review by jaclyn-taroni — APPROVED** [review]
I had a couple remaining questions, but for the most part this looks good to me! 👍  I'm going to label this with `don't merge` for now so @jharenza has the opportunity to review and I don't accidentally merge it before that happens.

File: analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd
Comment (kgaonkar6) [reply analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd:70]: right, I mean I know that LGAT doesn't have any WXS. I can make it more generic `lgat_dna`

File: analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd
Comment (kgaonkar6) [reply analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd]: correct!

File: analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd
Comment (kgaonkar6) [reply analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd:112]: It should only be NF-1, sorry I guess I misread

**Comment by kgaonkar6** [comment]
thanks for the review @jharenza ! I updated to LGG wildtype as per your condition of if DNA and RNA exists and no subtyping aberrations are found. And LGG, To be classified is DNA |RNA ids are missing. For NF1 I only updated the subtyping value as "NF1-germline" and "NF1-somatic"

This id has both NF1-somatic and NF1-germline
7316-2990	PT_1H2REHT2	BS_E0N3PTPN	BS_7M7JNG00	LGG, NF1-somatic, NF1-germline, CDKN2A/B

**Comment by jharenza** [comment]
>thanks for the review @jharenza ! I updated to LGG wildtype as per your condition of if DNA and RNA exists and no subtyping aberrations are found. And LGG, To be classified is DNA |RNA ids are missing. For NF1 I only updated the subtyping value as "NF1-germline" and "NF1-somatic"
>
>This id has both NF1-somatic and NF1-germline
7316-2990 PT_1H2REHT2 BS_E0N3PTPN BS_7M7JNG00 LGG, NF1-somatic, NF1-germline, CDKN2A/B

Perfect, I will just double check!

File: analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd
Comment (jharenza) [inline analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd:303]:
```suggestion
       !is.na(Kids_First_Biospecimen_ID_RNA)) ~ "LGG, wildtype",
```
Should just be wildtype, not H3 WT

**Review by jharenza — APPROVED** [review]
Just one typo fix that needs to be rerun, otherwise, I will approve and ask @jaclyn-taroni to review once you push that update.

File: analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd
Comment (kgaonkar6) [reply analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd]: Thanks for the check! I committed your suggestion and re-ran the script to compile the LGAT subtyping.

**Review by jharenza — APPROVED** [review]
Thanks for the updates! @jaclyn-taroni do you want to re-review or are we good to merge? Looks good on my end now.

File: analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd
Comment (jaclyn-taroni) [inline analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd:307]: Based on the comment I would expect the check to be `is.na()`, rather than `!is.na()`
```suggestion
    molecular_subtype=="NA" & (is.na(Kids_First_Biospecimen_ID_DNA) |
```

When those columns have `NA` values that means there is no sample, is that correct?

**Comment by jaclyn-taroni** [comment]
Had one question about some logic before merging 👍

File: analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd
Comment (kgaonkar6) [reply analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd:307]: that's right, let me update that.

**Pull request merged**
